src/fairness/writer/test/data_writer_db_test01.cpp

```diff
@@ -0,0 +1,138 @@
+/*****************************************************************************\
+ *  Copyright (c) 2021 Lawrence Livermore National Security, LLC.  Produced at
+ *  the Lawrence Livermore National Laboratory (cf, AUTHORS, DISCLAIMER.LLNS).
+ *  LLNL-CODE-658032 All rights reserved.
+ *
+ *  This file is part of the Flux resource manager framework.
+ *  For details, see https://github.com/flux-framework.
+ *
+ *  This program is free software; you can redistribute it and/or modify it
+ *  under the terms of the GNU General Public License as published by the Free
+ *  Software Foundation; either version 2 of the license, or (at your option)
+ *  any later version.
+ *
+ *  Flux is distributed in the hope that it will be useful, but WITHOUT
+ *  ANY WARRANTY; without even the IMPLIED WARRANTY OF MERCHANTABILITY or
+ *  FITNESS FOR A PARTICULAR PURPOSE.  See the terms and conditions of the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along
+ *  with this program; if not, write to the Free Software Foundation, Inc.,
+ *  59 Temple Place, Suite 330, Boston, MA 02111-1307 USA.
+ *  See also:  http://www.gnu.org/licenses/
+\*****************************************************************************/
+
+#if HAVE_CONFIG_H
+# include <config.h>
+#endif
+
+#include <cmath>
+#include <vector>
+#include <tuple>
+
+#include "src/fairness/reader/data_reader_db.hpp"
+#include "src/fairness/writer/data_writer_db.hpp"
+#include "src/common/libtap/tap.h"
+
+using namespace Flux::accounting;
+using namespace Flux::reader;
+using namespace Flux::writer;
+
+double fetch_fshare (sqlite3 *DB, const std::string &u, const std::string &b)
+{
+    std::string s;
+    sqlite3_stmt *cs = nullptr;
+
+    s = "SELECT fairshare FROM association_table WHERE username=? AND bank=?";
+
+    sqlite3_prepare_v2 (DB, s.c_str (), -1, &cs, 0);
+
+    sqlite3_bind_text (cs, 1, u.c_str (), -1, NULL);
+    sqlite3_bind_text (cs, 2, b.c_str (), -1, NULL);
+
+    sqlite3_step (cs);
+
```

Comment:
Not that it matters all that much since this is a test, but does the prepared statement need to have `finalize` called on it?

Reply:
Yes it does, thank you for pointing that out. I pushed up a `fixup` commit 0eb2e05 that calls `sqlite3_finalize` before returning the fairshare value.